lib/galaxy/tools/execute.py

```diff
@@ -72,6 +72,11 @@ def execute_single_job(execution_slice, completed_job):
         if job:
             log.debug(job_timer.to_str(tool_id=tool.id, job_id=job.id))
             execution_tracker.record_success(execution_slice, job, result)
+            # associate dataset instances with the job that creates them
+            if result:
+                datasets = [pair[1] for pair in result if type(pair[1]) == model.HistoryDatasetAssociation]
```

Comment:
These could also be LDDAs I think?

Reply:
Yes. Truth be told, I think I was so afraid of these edits breaking something, I was moving half a baby step at a time. The LDDAs here would be the second half. Let me complete that one :)

Reply:
Done.